token/program-2022/src/processor.rs

```diff
@@ -297,6 +298,11 @@ impl Processor {
             )?,
         };
 
+        if self_transfer || amount == 0 {
```

Comment:
wdyt about unconditionally checking as belts and suspenders?  Slightly more compute for the false case seems to be the only downside

Reply:
It depends on how sensitive we want to be about raising compute costs.  I'm a bit worried that the extensions will make things more expensive, and these owner checks are already quite pedantic.

How about this: we remove the checks now, and when we're ready to release, we revisit compute costs and see if the savings are worth it at that point?

Reply:
Deal!